src/ConnectedMode/Suppressions/ServerIssuesStore.cs

```diff
@@ -86,6 +87,21 @@ public void AddIssues(IEnumerable<SonarQubeIssue> issues, bool clearAllExistingI
             RaiseEventIfRequired();
         }
 
+        public void Reset()
+        {
+            if (!Get().Any())
```

Comment:
Unnecessary double lock acquisition. This condition can be removed completely or at least moved inside the lock below.